x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -325,6 +286,26 @@ static void getMemoryAndProcessors(
         );
     }
 
+    /**
+     * Return some autoscaling stats that tell the autoscaler not to change anything, but without making it think an error has occurred.
+     */
+    public static MlAutoscalingStats noScaleStats(ClusterState clusterState) {
+        int numberMlNodes = (int) clusterState.nodes().stream().filter(node -> node.getRoles().contains(DiscoveryNodeRole.ML_ROLE)).count();
+        return new MlAutoscalingStats(
```

Comment:
Just to confirm, these stats wouldn't accidentally tell the auto scaler to scale down would they?

Reply:
No, the autoscaler scales down when the penultimate value in the arguments here matches the size of an ML node. In this case it's zero.